Sources/GRPC/ConnectionBackoff.swift

```diff
@@ -135,7 +135,12 @@ public class ConnectionBackoffIterator: IteratorProtocol {
     case let .limited(limit) where limit > 0:
       self.connectionBackoff.retries.limit = .limited(limit - 1)
 
-    // limit must be <= 0, no new element.
+    // limit is reached, return an element with only a timeout.
+    case let .limited(limit) where limit == 0:
+      self.connectionBackoff.retries.limit = .limited(limit - 1)
+      return self.makeElement(backoff: nil)
```

Comment:
@glbrntt What do you think about using the Element.backoff as the sentinel value for the iterator?

Reply:
It changes the `Element` type which is an API breaking change so we can't do it unfortunately. Otherwise it'd be a good solution.